contracts/AccountTree.sol

```diff
@@ -24,24 +24,27 @@ contract AccountTree {
     bytes32[DEPTH] public filledSubtreesLeft;
     bytes32[DEPTH - BATCH_DEPTH] public filledSubtreesRight;
 
-    constructor() public {
+    constructor(
+        bytes32 initialRootLeft,
+        uint256 initislLeafIndexLeft,
+        bytes32[DEPTH] memory initialFilledSubtreesLeft
+    ) public {
         // prettier-ignore
         bytes32 firstZero = 0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563;
         // i = 0
         zeros[0] = firstZero;
-        filledSubtreesLeft[0] = firstZero;
         // i > 0
         for (uint256 i = 1; i < DEPTH; i++) {
             zeros[i] = keccak256(abi.encode(zeros[i - 1], zeros[i - 1]));
-            if (DEPTH > i) {
```

Comment:
Am I missing something or was this branch always taken? Looks wrong, but it's removed so doesn't matter anymore.

Reply:
Definitely always taken. It might have been added to satisfy a relatively dumb static analysis tool? This condition makes it easy to prove we never go out of bounds, slightly easier than understanding the for loop.